File: ruby/reverse_string.rb
require 'pry'
def reverse_string(str)
  # type your code in here

  # reversedString = ''
  # string_length = str.length - 1
  # string_length.downto(0).each do |l|
  #   reversedString << str[l]
  # end
  # reversedString

  str = str.split("")
  reversed = []
  str.each {|i| reversed.unshift(i)}
  reversed.join("")







if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: javascript/reverse_string.js
let str = "hi my name is nicholas"

function reverseString(str) {
  // type your code here
  let arr = str.split('')
  let reversed = []
  for (let i = arr.length; i >0; i --){
    reversed.push(arr[i])
  }
return reversed.join("")
 
  // return reversed
  // .join("")
}

reverseString('hi')



if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution

//create a new blank string
//use a for loop iterating over the letters of the original string, starting from
//the end and decreimenting, putting each letter into the new string in reverse
//order